src/components/MovieDetails/MovieDetails.jsx

```diff
@@ -8,23 +8,28 @@ class MovieDetails extends React.Component {
 
   render() {
     const movie = this.props.movieInfo;
-    const genresString = movie.genres.join();
+    let genresString;
+    if (typeof movie.genres === "undefined") {
```

Comment:
You can simplify this condition.

`if(!movie?.genres?.length) {...}` 

Moreover, you can use the ternary operator here (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator).

`const genresString = movie?.genres?.length ? movies.genres.join() : ''`